src/ntfs/read.rs

```diff
@@ -334,7 +334,8 @@ pub fn manage_stop_times(collections: &mut Collections, path: &path::Path) -> Re
     collections.stop_time_ids = stop_time_ids;
     let mut vehicle_journeys = collections.vehicle_journeys.take();
     for vj in &mut vehicle_journeys {
-        vj.stop_times.sort_unstable_by_key(|st| st.sequence);
+        vj.sort_and_check_stop_times()
+            .map_err(|e| format_err!("{}", e))?;
```

Comment:
Note that now that your new error type implements `std::error::Error`, you can do this

```suggestion
        vj.sort_and_check_stop_times()?;
```